helm/provider_test.go

```diff
@@ -44,36 +44,58 @@ func TestMain(m *testing.M) {
 		panic("Could not create temporary directory for helm config files")
 	}
 
-	os.Setenv("HELM_REPOSITORY_CONFIG", filepath.Join(home, "config/repositories.yaml"))
-	os.Setenv("HELM_REPOSITORY_CACHE", filepath.Join(home, "cache/helm/repository"))
-	os.Setenv("HELM_REGISTRY_CONFIG", filepath.Join(home, "config/registry.json"))
-	os.Setenv("HELM_PLUGINS", filepath.Join(home, "plugins"))
-	os.Setenv("XDG_CACHE_HOME", filepath.Join(home, "cache"))
+	err = os.Setenv("HELM_REPOSITORY_CONFIG", filepath.Join(home, "config/repositories.yaml"))
+	if err != nil {
+		panic("setenv failed for HELM_REPOSITORY_CONFIG")
+	}
+
+	err = os.Setenv("HELM_REPOSITORY_CACHE", filepath.Join(home, "cache/helm/repository"))
+	if err != nil {
+		panic("setenv failed for HELM_REPOSITORY_CACHE")
+	}
+
+	err = os.Setenv("HELM_REGISTRY_CONFIG", filepath.Join(home, "config/registry.json"))
+	if err != nil {
+		panic("setenv failed for HELM_REGISTRY_CONFIG")
+	}
+
+	err = os.Setenv("HELM_PLUGINS", filepath.Join(home, "plugins"))
+	if err != nil {
+		panic("setenv failed for HELM_PLUGINS")
+	}
+
+	err = os.Setenv("XDG_CACHE_HOME", filepath.Join(home, "cache"))
+	if err != nil {
+		panic("setenv failed for XDG_CACHE_HOME")
+	}
 
 	ec := m.Run()
 
-	os.RemoveAll(home)
+	err = os.RemoveAll(home)
+	if err != nil {
+		panic("TempDir deletion failed")
```

Comment:
Perhaps spit out the error to the user here - `TempDir deletion` doesn't really tell me too much. I'd like to know if it was a permission denied error, for example.

Reply:
Just realized this is test code - I would just panic and print out the `err`.

Reply:
Good idea. I'll actually do that for all the others in that file too. I didn't realize panic() could print an error message, but it turns out it can. https://gobyexample.com/panic